src/WalkingModule/src/Module.cpp

```diff
@@ -1751,3 +1900,100 @@ bool WalkingModule::stopWalking()
+        {
+            if (m_rightInContact.at(0))
+            {
+                stanceFoot = "right";
+                footTransformToWorld = m_FKSolver->getRightFootToWorldTransform();
+            }
+            else
+            {
+                stanceFoot = "left";
+                footTransformToWorld = m_FKSolver->getLeftFootToWorldTransform();
+            }
+            inContact = true;
+        }
+        else
+        {
+            inContact = false;
+        }
+        root_linkTransform = m_FKSolver->getRootLinkToWorldTransform();   //world -> rootLink transform
+        //Data conversion and port writing
+        if (inContact)
+        {
+            if (m_trajectoryGenerator->getUnicycleState(virtualUnicyclePose, virtualUnicycleReference, stanceFoot))
+            {
+                //send data
+                yarp::os::Stamp stamp (0, yarp::os::Time::now());   //move to private member of the class
+                m_unicyclePort.setEnvelope(stamp);
+                auto& data = m_unicyclePort.prepare();
+                data.clear();
+                auto& unicyclePose = data.addList();
+                unicyclePose.addFloat64(virtualUnicyclePose(0));
+                unicyclePose.addFloat64(virtualUnicyclePose(1));
+                unicyclePose.addFloat64(virtualUnicyclePose(2));
+                auto& referencePose = data.addList();
+                referencePose.addFloat64(virtualUnicycleReference(0));
+                referencePose.addFloat64(virtualUnicycleReference(1));
+                referencePose.addFloat64(virtualUnicycleReference(2));
+                data.addString(stanceFoot);
+
+                //add root link stransform: X, Y, Z, r, p, yaw,
+                auto& transform = data.addList();
+                transform.addFloat64(root_linkTransform.getPosition()(0));
+                transform.addFloat64(root_linkTransform.getPosition()(1));
+                transform.addFloat64(root_linkTransform.getPosition()(2));
+                transform.addFloat64(root_linkTransform.getRotation().asRPY()(0));
+                transform.addFloat64(root_linkTransform.getRotation().asRPY()(1));
+                transform.addFloat64(root_linkTransform.getRotation().asRPY()(2));
+
+                //planned velocity of the CoM
+                auto comVel = m_stableDCMModel->getCoMVelocity();
+                auto& velData = data.addList();
+                velData.addFloat64(comVel(0));
+                velData.addFloat64(comVel(1));
+                m_unicyclePort.write();
+            }
+            else
+            {
+                yError() << "[WalkingModule::computeVirtualUnicycleThread] Could not getUnicycleState from m_trajectoryGenerator (no data sent).";
+            }
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000/m_navigationTriggerLoopRate));
+    }
+    yInfo() << "[WalkingModule::computeVirtualUnicycleThread] Terminated thread";
+}
```

Comment:
As mentioned in https://github.com/robotology/walking-controllers/pull/144 it would be best if the modifications to ``Module.cpp`` were minimal. This file is meant to connect all the different bits and pieces. It is more of a coordinator, so it should have limited logic.